shell/platform/darwin/ios/ios_external_texture_gl.mm

```diff
@@ -55,10 +66,82 @@
+
+  {
+    CVOpenGLESTextureRef uvTexture;
+    CVReturn err = CVOpenGLESTextureCacheCreateTextureFromImage(
+        kCFAllocatorDefault, cache_ref_, buffer_ref_, nullptr, GL_TEXTURE_2D, GL_LUMINANCE_ALPHA,
+        width / 2, height / 2, GL_LUMINANCE_ALPHA, GL_UNSIGNED_BYTE, 1, &uvTexture);
+    if (err != noErr) {
+      FML_DCHECK(uvTexture) << "Could not create texture from pixel buffer: " << err;
+    } else {
+      uv_texture_ref_.Reset(uvTexture);
+    }
+  }
+}
+
+sk_sp<SkImage> IOSExternalTextureGL::CreateImageFromRGBATexture(GrContext* context,
+                                                                const SkRect& bounds) {
+  GrGLTextureInfo textureInfo = {CVOpenGLESTextureGetTarget(texture_ref_),
+                                 CVOpenGLESTextureGetName(texture_ref_), GL_RGBA8_OES};
+  GrBackendTexture backendTexture(bounds.width(), bounds.height(), GrMipMapped::kNo, textureInfo);
+  sk_sp<SkImage> image =
+      SkImage::MakeFromTexture(context, backendTexture, kTopLeft_GrSurfaceOrigin,
+                               kRGBA_8888_SkColorType, kPremul_SkAlphaType, nullptr);
+  return image;
+}
+
+sk_sp<SkImage> IOSExternalTextureGL::CreateImageFromYUVTextures(GrContext* context,
+                                                                const SkRect& bounds) {
+  GrGLTextureInfo yTextureInfo = {CVOpenGLESTextureGetTarget(y_texture_ref_),
+                                  CVOpenGLESTextureGetName(y_texture_ref_), GR_GL_LUMINANCE8};
+  GrBackendTexture yBackendTexture(bounds.width(), bounds.height(), GrMipMapped::kNo, yTextureInfo);
+  GrGLTextureInfo uvTextureInfo = {CVOpenGLESTextureGetTarget(uv_texture_ref_),
+                                   CVOpenGLESTextureGetName(uv_texture_ref_), GR_GL_RGBA8};
+  GrBackendTexture uvBackendTexture(bounds.width(), bounds.height(), GrMipMapped::kNo,
+                                    uvTextureInfo);
+  GrBackendTexture nv12TextureHandles[] = {yBackendTexture, uvBackendTexture};
+  SkYUVAIndex yuvaIndices[4] = {
+      SkYUVAIndex{0, SkColorChannel::kR},  // Read Y data from the red channel of the first texture
+      SkYUVAIndex{1, SkColorChannel::kR},  // Read U data from the red channel of the second texture
+      SkYUVAIndex{
+          1, SkColorChannel::kA},  // Read V data from the alpha channel of the second texture,
+                                   // normal NV12 data V should be taken from the green channel, but
+                                   // currently only the uv texture created by GL_LUMINANCE_ALPHA
+                                   // can be used, so the V value is taken from the alpha channel
+      SkYUVAIndex{-1, SkColorChannel::kA}};  //-1 means to omit the alpha data of YUVA
+  SkISize size{yBackendTexture.width(), yBackendTexture.height()};
+  sk_sp<SkImage> image =
+      SkImage::MakeFromYUVATextures(context, kRec601_SkYUVColorSpace, nv12TextureHandles,
+                                    yuvaIndices, size, kTopLeft_GrSurfaceOrigin, nullptr);
```

Comment:
nit: nullptr argument would benefit from an argument name comment

Reply:
This is also to keep the style consistent with the original code. Do I need to modify the original code according to your suggestions?

Reply:
Yep, https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

Reply:
done